packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -145,6 +145,109 @@ const traverseStyledBinaryExpression = (node: t.BinaryExpression, nestedVisitor:
   return node;
 };
 
+/**
+ * Handles cases like:
+ * 1. `propz.loading` in `border-color: \${(propz) => (propz.loading ? colors.N100 : colors.N200)};`
+ * Outcome: It will replace `propz.loading` with `props.loading`.
+ *
+ * 2. `prop.notValidProp` in `border-color: \${(props) => (props.notValidProp ? colors.N100 : colors.N200)};`
```

Comment:
what is the edge case here? that the arg is `props` but we're using `prop`?

in this case should we just do nothing?

Reply:
oops..it should be `props.notValidProp`. Sorry 😅